server/database/migrations/2016_07_25_203547_mark_note_replies_as_notes.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+use App\Models\Annotation;
+
+class MarkNoteRepliesAsNotes extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        $annotations = DB::table('annotations')->whereNotNull('data')->get();
+        foreach ($annotations as $annotation) {
+            $data = json_decode($annotation->data, true);
+
+            if (empty($data['old_permalink_type']) || $data['old_permalink_type'] !== 'annsubcomment') {
```

Comment:
`data` is not an actual JSON column type, just JSON is stored as a text column. We would have to require MySQL >= 5.7 and Laravel >= 5.2.23 in order to use JSON column types. We could use a `LIKE` expression to get just the records that have `annsubcomment` in the text, but in order to extract the old id it still seems the best to just decode the JSON and do the work in PHP unfortunately.